docs/articles/documentation/test-api/intercepting-and-mocking-http-requests.md

```diff
@@ -0,0 +1,547 @@
+  * [Filtering by a URL](#filtering-by-a-url)
+  * [Filtering by a Regular Expression](#filtering-by-a-regular-expression)
+  * [Filtering by Request Parameters](#filtering-by-request-parameters)
+  * [Filtering by a Predicate](#filtering-by-a-predicate)
+* [Attaching Hooks to Tests and Fixtures](#attaching-hooks-to-tests-and-fixtures)
+* [Creating a Custom HTTP Request Hook](#creating-a-custom-http-request-hook)
+  * [Understanding How TestCafe Request Hooks Operate](#understanding-how-testcafe-request-hooks-operate)
+  * [Writing a Hook](#writing-a-hook)
+
+## Logging HTTP Requests
+
+To log HTTP requests sent during test execution, use the `RequestLogger`. This object stores all requests sent and responses received while the test is running. You can use this information in [assertions](assertions/README.md) to check how the tested page communicates with HTTP services.
+
+To create a request logger, use the `RequestLogger` constructor.
+
+```text
+RequestLogger([filter] [, options])
+```
+
+Parameter    | Type | Description | Default
+------------ | ---- | ----------- | --------
+`filter`&#160;*(optional)*  | String &#124; RegExp &#124; Object &#124; Predicate | Specifies which requests should be tracked by the logger. See [Specifying Which Requests are Handled by the Hook](#specifying-which-requests-are-handled-by-the-hook). | All requests are tracked
+`options`&#160;*(optional)* | Object | Options that define how the requests and responses are logged. | See below
+
+The `options` parameter contains the following options.
+
+Option | Type | Description   | Default
+------ | ---- | ------------- | ---------
+`logRequestHeaders` | Boolean | Specifies whether the request headers should be logged. | `false`
+`logRequestBody` | Boolean | Specifies whether the request body should be logged. | `false`
+`stringifyRequestBody` | Boolean | Specifies whether the request body should be stored as a String or a [Buffer](https://nodejs.org/api/buffer.html). | `false`
+`logResponseHeaders` | Boolean | Specifies whether the response headers should be logged. | `false`
+`logResponseBody` | Boolean | Specifies whether the response body should be logged. | `false`
+`stringifyResponseBody` | Boolean | Specifies whether the response body should be stored as a string or a [Buffer](https://nodejs.org/api/buffer.html). | `false`
+
+```js
+import { RequestLogger } from 'testcafe';
+
+const simpleLogger = RequestLogger('http://example.com');
+const headerLogger = RequestLogger(/testcafe/, {
+    logRequestHeaders: true,
+    logResponseHeaders: true
+});
+```
+
+To enable the logger to track the requests, [attach it to a test or fixture](#attaching-hooks-to-tests-and-fixtures).
+
+By default, `RequestLogger` stores the following parameters.
+
+* The URL where the request is sent.
+* The request's HTTP method.
+* The status code received in the response.
+* The user agent that sent the request.
+* The ID of the test run that sent the request.
```

Comment:
It needs only for customers which will create an own hook. I think we should not describe it.